src/hooks/useNotifications.js

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from 'react'
+import messaging from '@react-native-firebase/messaging'
+
+import { useClient } from 'cozy-client'
+import { removeNotificationDeviceToken } from '/libs/client'
+import {
+  handleNotificationTokenReceiving,
+  handleNotificationOpening,
+  requestAndGetNotificationPermission
+} from '/libs/notifications/notifications'
+
+export const useNotifications = () => {
```

Comment:
This got some logic, we should extract it as pure functions and write easy unit tests for it.
Also as a new file, it should be pure TS if you are able to use it

Reply:
It is hard to separate more because we have some React setXXX.

But I migrated to ts (and it helped to spot some issues).